radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/TransitionedParticle.java

```diff
@@ -0,0 +1,65 @@
+package com.radixdlt.client.core.ledger;
+
+import com.radixdlt.client.core.atoms.AtomObservation;
+import com.radixdlt.client.core.atoms.AtomObservation.Type;
+import com.radixdlt.client.core.atoms.particles.Particle;
+import com.radixdlt.client.core.atoms.particles.Spin;
+import com.radixdlt.client.core.atoms.particles.SpunParticle;
+
+/**
+ * A transitioned particle represents an action/instruction which higher layer protocols use
+ * to update their state. It differs from a SpunParticle in that TransitionedParticles include
+ * the ability to revert previous SpunParticles, thus the particle state machine complexity
+ * increases by a factor of exactly 2.
+ *
+ * @param <T> the type of particle
+ */
+public class TransitionedParticle<T extends Particle> {
+	private final ParticleTransition transition;
+	private final T particle;
+
+	public TransitionedParticle(ParticleTransition transition, T particle) {
+		this.transition = transition;
+		this.particle = particle;
+	}
+
+	public T getParticle() {
+		return particle;
+	}
+
+	public Spin getSpinTo() {
+		return transition.getSpinTo();
+	}
+
+	public Spin getSpinFrom() {
+		return transition.getSpinFrom();
+	}
+
+	public ParticleTransition getTransition() {
+		return transition;
+	}
+
+	public static <T extends Particle> TransitionedParticle<T> fromSpunParticle(
+		SpunParticle<T> s, AtomObservation.Type observationType
+	) {
+		final boolean revert = observationType == Type.DELETE;
+		ParticleTransition transition = ParticleTransition.to(s.getSpin(), revert);
+		return new TransitionedParticle<>(transition, s.getParticle());
+	}
+
+	public static <T extends Particle> TransitionedParticle<T> n2u(T particle) {
```

Comment:
Same comment from further down, I think the spelt out versions (e.g. `neutralToDown`) would be more readable. I appreciate that the current version looks cooler though ;)